core/src/main/scala/pickling/Macros.scala

```diff
@@ -31,6 +31,25 @@ trait TypeAnalysis extends Macro {
   }
 }
 
+// purpose of this macro: a wrapper around both pickler and unpickler
+trait PicklerUnpicklerMacros extends Macro {
+  import c.universe._
+
+  def impl[T: c.WeakTypeTag]: c.Tree = preferringAlternativeImplicits {
+    val originalTpe = weakTypeOf[T]
+    // Note: this makes it so modules work, things like foo.type.
+    //       For some reason we get an issue with not having == defined on Class[_] otherwise.
+    val tpe = {
+      // TODO - fix this for certain primitive types, like Null, etc.
```

Comment:
If I think about it widening happens in `SPickler.generate` anyways so I should be able to remove this pasta.
It seems to work locally:

```
scala> Foo.pickle // Foo is a companion object
res2: scala.pickling.json.pickleFormat.PickleType =
JSONPickle({
  "tpe": "Foo.type"
})
```